### ISSUE TITLE
Feature to remove javax.annotation.Generated annotation

### ISSUE THREAD

**Issue by bjakke** [immutables/immutables#566]
In `value-processor/src/org/immutables/value/processor/Immutables.generator` there is:
```
[if classpath.available 'javax.annotation.Generated']
@javax.annotation.Generated({"Immutables.generator", "[type.typeAbstract.relative]"})
[/if]
```
Could it be possible to create a styling option (by default keeping existing way) to always disable generation of this annotation even when it is in classpath?

The use-case: One party creates sources where it is available, another party imports them to android studio -> code does not compile as the javax.annotation.Generated annotation is not available in Android. I know there are possible workarounds by adding a dependency or the first party might remove the annotation as manual replacement, but it would be nice to have this option directly.

**Comment by elucash** [comment]
As this and some other functionality relies on classpath lookup and availability of certain classes, the solution was generalized to specify list of blacklisted classpath prefixes so the mentioned classpath availability checks would not find them, resulting in generated code would not use features related to missing classes (like adding optional `javax.annotation.Generated` annotations).

For this to work, META-INF mechanism was utilized rather than styling options. Add unwanted fqn prefixes as the lines to the `META-INF/extensions/org.immutables.inhibit-classpath` resource file available in classpath, see example file: https://github.com/immutables/immutables/blob/master/android-stub/test/META-INF/extensions/org.immutables.inhibit-classpath

**Comment by elucash** [comment]
released in 2.4.4. thank you!
